Tests/UnitTests/Core/Detector/BlueprintHelperTests.cpp

```diff
@@ -0,0 +1,285 @@
+  BOOST_CHECK(detector->children[0u]->name == "beam_pipe");
+  BOOST_CHECK(detector->children[1u]->name == "pixel");
+
+  // Now fill the gaps
+  Acts::Experimental::detail::BlueprintHelper::fillGaps(*detector);
+
+  // Do the tests again
+  BOOST_CHECK(detector->children.size() == 4u);
+  BOOST_CHECK(detector->children[0u]->name == "beam_pipe");
+  BOOST_CHECK(detector->children[1u]->name == "detector_gap_0");
+  BOOST_CHECK(detector->children[2u]->name == "pixel");
+  BOOST_CHECK(detector->children[3u]->name == "detector_gap_1");
+
+  // Adjustment of gap parameters
+  BOOST_CHECK(detector->children[1u]->boundaryValues[0] == beamPipeOr);
+  BOOST_CHECK(detector->children[1u]->boundaryValues[1] == pixelIr);
+  BOOST_CHECK(detector->children[1u]->boundaryValues[2] == detectorHz);
+
+  BOOST_CHECK(detector->children[3u]->boundaryValues[0] == pixelOr);
+  BOOST_CHECK(detector->children[3u]->boundaryValues[1] == detectorOr);
+  BOOST_CHECK(detector->children[3u]->boundaryValues[2] == detectorHz);
+
+  // Check the pixel system: Nec / Barrel / Pec
+  BOOST_CHECK(detector->children[2u]->children.size() == 3u);
+  BOOST_CHECK(detector->children[2u]->children[0u]->children.size() == 3u);
+  BOOST_CHECK(detector->children[2u]->children[1u]->children.size() == 5u);
+  BOOST_CHECK(detector->children[2u]->children[2u]->children.size() == 3u);
+
+  // Nec test
+  BOOST_CHECK(
+      detector->children[2u]->children[0u]->children[0]->boundaryValues[0] ==
+      pixelIr);
+  BOOST_CHECK(
+      detector->children[2u]->children[0u]->children[0]->boundaryValues[1] ==
+      pixelOr);
+
+  BOOST_CHECK(
+      detector->children[2u]->children[0u]->children[1]->boundaryValues[0] ==
+      pixelIr);
+  BOOST_CHECK(
+      detector->children[2u]->children[0u]->children[1]->boundaryValues[1] ==
+      pixelOr);
+
+  BOOST_CHECK(
+      detector->children[2u]->children[0u]->children[2]->boundaryValues[0] ==
+      pixelIr);
+  BOOST_CHECK(
+      detector->children[2u]->children[0u]->children[2]->boundaryValues[1] ==
+      pixelOr);
+
+  std::ofstream fs2("detector_without_gaps.dot");
+  detector->dotStream(fs2);
+  fs2.close();
+}
+
+BOOST_AUTO_TEST_CASE(BlueprintCylindricalGapException) {
+  auto innerBuilder =
+      std::make_shared<Acts::Experimental::IInternalStructureBuilder>();
+
+  // The root node - detector
+  std::vector<Acts::ActsScalar> detectorBoundaries = {0., 50., 100.};
+  std::vector<Acts::BinningValue> detectorBinning = {Acts::binX};
+  auto detector = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "detector", Acts::Transform3::Identity(), Acts::VolumeBounds::eCuboid,
+      detectorBoundaries, detectorBinning);
+
+  std::vector<Acts::ActsScalar> cubeOneBoundaries = {0., 20., 100.};
+  auto cubeOne = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "cubeOne", Acts::Transform3::Identity(), Acts::VolumeBounds::eCuboid,
+      detectorBoundaries, innerBuilder);
+  detector->add(std::move(cubeOne));
+
+  BOOST_CHECK_THROW(
```

Comment:
Is that the only check? E.g. is there a graph printed for this?

Reply:
Na, it fails, it will not print a graph - but I added the only other exception that could be thrown.

Reply:
Is this failure expected or will it be fixed in the followup PR?